domain/src/main/java/org/oppia/domain/question/QuestionTrainingController.kt

```diff
@@ -35,14 +33,36 @@ class QuestionTrainingController @Inject constructor(
    */
    fun startQuestionTrainingSession(skillIdsList: List<String>): LiveData<AsyncResult<Any?>> {
     return try {
-        val questionsList = retrieveQuestionsForSkillIds(skillIdsList)
-        questionAssessmentProgressController.beginQuestionTrainingSession(questionsList)
+        questionAssessmentProgressController.beginQuestionTrainingSession(
+          retrieveQuestionsForSkillIds(skillIdsList))
         MutableLiveData(AsyncResult.success<Any?>(null))
```

Comment:
Should we be returning the data provider-converted-live data here? That would provide a signal to the UI for when the training session can begin.

Reply:
Wouldn't that mean that question assessment progress controller would need to attach an observer? I thought we wanted to pass the data provider itself to the assessment controller?

Reply:
Ahh okay I misunderstood, I've changed this